mat64/matrix.go

```diff
@@ -328,6 +322,77 @@ func Det(a Matrix) float64 {
 	return math.Exp(det) * sign
 }
 
+// Equal returns whether element a and b have the same size and contain all equal
+// elements.
+func Equal(a, b Matrix) bool {
+	ar, ac := a.Dims()
```

Comment:
It's actially worse that that. If a user defined Matrix has a concrete type base and the type is of contains a sloce, map, chan or func, the test would panic since that are not comparable. Those are unlikely, but not impossible.